hibernate-core/src/test/java/org/hibernate/orm/test/query/criteria/CriteriaMultiselectGroupByAndOrderByTest.java

```diff
@@ -72,30 +81,104 @@ public void tearDown(SessionFactoryScope scope) {
 	}
 
 	@Test
+	@Jira( "https://hibernate.atlassian.net/browse/HHH-17085" )
 	public void testCriteriaGroupBy(SessionFactoryScope scope) {
-		executeQuery( scope, false );
+		executeQuery( scope, false, false );
 	}
 
 	@Test
+	@Jira( "https://hibernate.atlassian.net/browse/HHH-17085" )
 	public void testCriteriaGroupByAndOrderBy(SessionFactoryScope scope) {
-		executeQuery( scope, true );
+		executeQuery( scope, true, false );
+	}
+
+	@Test
+	@Jira( "https://hibernate.atlassian.net/browse/HHH-17085" )
+	public void testCriteriaGroupByAndOrderByAndHaving(SessionFactoryScope scope) {
+		executeQuery( scope, true, true );
 	}
 
-	private void executeQuery(SessionFactoryScope scope, boolean order) {
+	private void executeQuery(SessionFactoryScope scope, boolean order, boolean having) {
 		scope.inTransaction( session -> {
 			final CriteriaBuilder cb = session.getCriteriaBuilder();
 			final CriteriaQuery<Tuple> query = cb.createQuery( Tuple.class );
 			final Root<Primary> root = query.from( Primary.class );
 			final Join<Primary, Secondary> join = root.join( "secondary" );
+			final Path<String> entityName = join.get( "entityName" );
+			final Expression<Number> sum = cb.sum( root.get( "amount" ) );
 			query.multiselect(
-					join.get( "entityName" ).alias( "secondary" ),
-					cb.sum( root.get( "amount" ) ).alias( "sum" )
+					entityName.alias( "secondary_name" ),
+					sum.alias( "amount_sum" )
 			).groupBy( join );
 			if ( order ) {
-				query.orderBy( cb.desc( join.get( "entityName" ) ) );
+				query.orderBy( cb.desc( entityName ) );
+			}
+			if ( having ) {
+				query.having( cb.and(
+						cb.equal( entityName, "a" ),
+						cb.gt( sum, 0 )
+				) );
 			}
 			final List<Tuple> resultList = session.createQuery( query ).getResultList();
-			assertThat( resultList ).hasSize( 3 );
+			assertThat( resultList ).hasSize( having ? 1 : 3 );
+		} );
+	}
+
+	@Test
+	@Jira( "https://hibernate.atlassian.net/browse/HHH-17231" )
+	public void testSubqueryGroupBy(SessionFactoryScope scope) {
+		executeSubquery( scope, false, false );
+	}
+
+	@Test
+	@Jira( "https://hibernate.atlassian.net/browse/HHH-17231" )
+	@SkipForDialect( dialectClass = SQLServerDialect.class, reason = "The ORDER BY clause is invalid derived tables, used for tuple in-subquery emulation")
```

Comment:
Just a side note, this is only invalid if you do not also use `limit`/`offset` because ordering in subqueries is meaningless without that. I think Sybase doesn't support `limit`/`offset` in subqueries though, so it's fine to skip that, but IMO the reason should be changed to reflect this fact.

Reply:
Right, I've added an `offset 0` and SQLServer works. I've kept the skip for sybase and updated the reason.